### PR TITLE
New module:  support Alibaba Cloud ECS (cloud/alicloud/ecs)

### PR DESCRIPTION
**Pull request by xiaozhu36**
SUMMARY

Added a new module alicloud ecs.py that can manage alicloud ecs instance.

COMPONENT NAME

lib/ansible/module_utils/alicloud_ecs.py
lib/ansible/modules/cloud/alicloud/ecs.py

ANSIBLE VERSION

2.4.0

### CODE REVIEW

**Comment by xiaozhu36** [comment]
Hello @sivel 

According to the last PR feedback, I have pulled new request about alicloud module, but there is always an error: No RETURN provided, but I noticed that the 'RETURN' is optional, could you please give me a hand to avoid the error? Thanks.

Regards,
Guimin

**Comment by sivel** [comment]
RETURN is required for all new modules

File: shippable.yml [hunk 1]
```diff
@@ -83,11 +83,11 @@ integrations:
       on_failure: always
       on_start: never
       on_pull_request: always
-    - integrationName: slack
```
Comment (gundalow) [inline shippable.yml]: Please revert the change to this file

File: lib/ansible/modules/cloud/alicloud/ecs.py [hunk 2]
```diff
@@ -0,0 +1,1213 @@
...
+        - A list of hash/dictionaries of instance attributes. If it's set, the specified instance's attributes would be modified.
+        - keys allowed are
+            - id (required=true; default=null; description=ID of an ECS instance )
+            - name (required=false; default=null; description=Name of the instance to modify)
+            - description (required=false; default=null; description=Description of the instance to use)
+            - password (required=false; default=null; description=The password to login instance)
+            - host_name (required=false; default=null; description=Instance host name)
+      required: false
+      default: null
+      aliases: []
```
Comment (gundalow) [inline lib/ansible/modules/cloud/alicloud/ecs.py]: Please remove empty `aliases:` lines

File: lib/ansible/modules/cloud/alicloud/ecs.py [hunk 3]
```diff
@@ -0,0 +1,1213 @@
...
+      description: The number of the new instance.
+      required: false
+      default: 1
+      aliases: []
+    allocate_public_ip:
+      description: Whether allocate a public ip for the new instance.
+      required: false
+      default: true
+      aliases: [ 'assign_public_ip' ]
+      choices: ["true", "false"]
```
Comment (gundalow) [inline lib/ansible/modules/cloud/alicloud/ecs.py]: Please replace this line with
`type: bool`

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by xiaozhu36** [comment]
Ok, thank you for your feedback. By the way, which version from start to require RETURN for all modules?

**Comment by sivel** [comment]
RETURN has been required for all new modules since 2.1, specifically it has been in CI enforcement since December 22, 2015.

File: shippable.yml (hunk 1 above)
Comment (xiaozhu36) [reply shippable.yml]: I'm very sorry, it's my fault. In order to make shippable test in my repository conveniently, I changed them, and I forgot to revert them when I pull new request. I will revert them in the next PR.

**Comment by xiaozhu36** [comment]
Thank you very much.

**Pull request closed without merging**